tests/tools/onert_train/src/rawdataloader.cc

```diff
@@ -45,33 +79,24 @@ Generator RawDataLoader::loadData(const std::string &input_file, const std::stri
     start += (bufsize_for(&expected_infos[i]) / batch_size * data_length);
   }
 
-  try
-  {
-    _input_file = std::ifstream(input_file, std::ios::ate | std::ios::binary);
-    _expected_file = std::ifstream(expected_file, std::ios::ate | std::ios::binary);
-  }
-  catch (const std::exception &e)
-  {
-    std::cerr << e.what() << std::endl;
-    std::exit(-1);
-  }
-
-  return [input_origins, expected_origins, &input_infos, &expected_infos,
-          this](uint32_t idx, std::vector<Allocation> &inputs, std::vector<Allocation> &expecteds) {
-    for (uint32_t i = 0; i < input_infos.size(); ++i)
-    {
-      auto bufsz = bufsize_for(&input_infos[i]);
-      _input_file.seekg(input_origins[i] + idx * bufsz, std::ios::beg);
-      _input_file.read(reinterpret_cast<char *>(inputs[i].data()), bufsz);
-    }
-    for (uint32_t i = 0; i < expected_infos.size(); ++i)
-    {
-      auto bufsz = bufsize_for(&expected_infos[i]);
-      _expected_file.seekg(expected_origins[i] + idx * bufsz, std::ios::beg);
-      _expected_file.read(reinterpret_cast<char *>(expecteds[i].data()), bufsz);
-    }
-    return true;
-  };
+  return std::make_tuple(
+    [input_origins, expected_origins, &input_infos, &expected_infos,
+     this](uint32_t idx, std::vector<Allocation> &inputs, std::vector<Allocation> &expecteds) {
+      for (uint32_t i = 0; i < input_infos.size(); ++i)
+      {
+        auto bufsz = bufsize_for(&input_infos[i]);
+        _input_file.seekg(input_origins[i] + idx * bufsz, std::ios::beg);
+        _input_file.read(reinterpret_cast<char *>(inputs[i].data()), bufsz);
+      }
+      for (uint32_t i = 0; i < expected_infos.size(); ++i)
+      {
+        auto bufsz = bufsize_for(&expected_infos[i]);
+        _expected_file.seekg(expected_origins[i] + idx * bufsz, std::ios::beg);
+        _expected_file.read(reinterpret_cast<char *>(expecteds[i].data()), bufsz);
+      }
+      return true;
+    },
+    data_length);
```

Comment:
I am not sure it is safe to return lambda to caller using `std::make_tuple`. `std::make_tuple` uses reference. If all variables from your lambda (e.g. captured variables of your lambda (`input_infos` and `expected_infos`)) and so on... are guaranteed to outlive than the receiver of `std::make_tuple`. It may be okay. However, it is hard to make sure manually by reading code. Static code analyzer may help.

Reply:
> Thus, all variables from your lambda (e.g. captured variables of your lambda (`input_infos` and `expected_infos`...) needs to be guaranteed to outlive than the receiver of `std::make_tuple`.

Yes, the captured variables is guaranteed to live outside this function because it is located in the same location as the caller function. I don't know exactly how to improve the readability of this code, but I will make some changes.

Reply:
@jyoungyun I am not saying the captured variables what I've mentioned above. I already checked it. My concern is there may be more captured stuff under hood, which I am missing. If you're sure, I am okay.